sdk/storage/inc/common/xml_wrapper.hpp

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+// TODO: remove this
+#pragma warning(disable : 4239)
+
+#include <libxml/xmlreader.h>
+#include <libxml/xmlwriter.h>
+#include <string>
+#include <stack>
+#include <memory>
+
+namespace Azure
+{
+namespace Storage
+{
+namespace Common
+{
+namespace XML
+{
+
+std::string xml_char_to_string(const xmlChar* xml_char);
+
+/// <summary>
+/// A class to wrap xmlTextReader of c library libxml2. This class provides abilities to read from
+/// xml format texts.
+/// </summary>
+class xml_text_reader_wrapper
+{
+public:
+  xml_text_reader_wrapper(const unsigned char* buffer, unsigned int size);
+
+  ~xml_text_reader_wrapper();
+
+  /// <summary>
+  /// Moves to the next node in the stream.
+  /// </summary>
+  /// <returns>true if the node was read successfully and false if there are no more nodes to
+  /// read</returns>
+  bool read();
+
+  /// <summary>
+  /// Gets the type of the current node.
+  /// </summary>
+  /// <returns>A integer that represent the type of the node</returns>
+  unsigned get_node_type();
+
+  /// <summary>
+  /// Checks if current node is empty
+  /// </summary>
+  /// <returns>True if current node is empty and false if otherwise</returns>
+  bool is_empty_element();
+
+  /// <summary>
+  /// Gets the local name of the node
+  /// </summary>
+  /// <returns>A string indicates the local name of this node</returns>
+  std::string get_local_name();
+
+  /// <summary>
+  /// Gets the value of the node
+  /// </summary>
+  /// <returns>A string value of the node</returns>
+  std::string get_value();
+
+  /// <summary>
+  /// Moves to the first attribute of the node.
+  /// </summary>
+  /// <returns>True if the move is successful, false if empty</returns>
+  bool move_to_first_attribute();
+
+  /// <summary>
+  /// Moves to the next attribute of the node.
+  /// </summary>
+  /// <returns>True if the move is successful, false if empty</returns>
+  bool move_to_next_attribute();
+
+private:
+  xmlTextReaderPtr m_reader;
```

Comment:
I don't know where this type comes from but if it's a pointer you probably don't want this thing to be copyable or assignable? Could this be `unique_ptr`?

Reply:
Ditto many places here.